src/js/modal.js

```diff
@@ -0,0 +1,156 @@
+    form.classList.add('content');
+
+    const h1 = document.createElement('h1');
+    h1.textContent = `Book room ${challenge.title} (step 2)`;
+    form.appendChild(h1);
+
+    const labelName = document.createElement('label');
+    labelName.textContent = 'Name';
+    labelName.for = 'input-name';
+    form.appendChild(labelName);
+   
+    const inputName = document.createElement('input');
+    inputName.type = 'text';
+    inputName.classList.add('input-name');
+    form.appendChild(inputName);
+
+    const labelEmail = document.createElement('label');
+    labelEmail.textContent = 'E-mail';
+    labelEmail.for = 'input-email';
+    form.appendChild(labelEmail);
+
+    const inputEmail = document.createElement('input');
+    inputEmail.type = 'text';
+    inputEmail.classList.add('input-email');
+    form.appendChild(inputEmail);
+
+    const labelTime = document.createElement('label');
+    labelTime.textContent = 'What time?';
+    labelEmail.for = 'select-time';
+    form.appendChild(labelTime);
+
+    const selectTime = document.createElement('select');
+    selectTime.classList.add('select-time');
+    form.appendChild(selectTime);
+
+    for (let i = 0; i < times.slots.length; i++) {
+        const option = document.createElement('option');
+        option.textContent = times.slots[i];
+        selectTime.appendChild(option);
+    }
+
+    const labelParticipants = document.createElement('label');
+    labelParticipants.textContent = 'How many participants?';
+    labelParticipants.for = 'select-participants';
+    form.appendChild(labelParticipants);
+
+    const selectParticipants = document.createElement('select');
+    selectParticipants.classList.add('select-participants');
+    form.appendChild(selectParticipants);
+
+    let amountOfOptions = challenge.maxParticipants - challenge.minParticipants;
+    for (let i = 0; i < amountOfOptions; i++) {
+        const option = document.createElement('option');
+        option.textContent = `${challenge.minParticipants + i} participants`;
+        selectParticipants.appendChild(option);
+    }
+
+    const button = document.createElement('button');
+    button.type = 'submit';
+    button.classList.add('button', 'primary');
+    button.name = 'submit-booking';
+    button.textContent = 'Submit booking';
+    form.appendChild(button);
+
+    
+    form.addEventListener('submit', async e => {
+        e.preventDefault();
+
+        await reserveTime(); //Martas function här
+
+        renderContent(step3());
+    });
+
+    return form;
+}
+
+function step3() {
+    const div = document.createElement('div');
+    div.classList.add('content');
+
+    const message = document.createElement('h1');
+    message.textContent = 'Thank you!';
+    div.appendChild(message);
+
+    const button = document.createElement('button');
+    button.classList.add('button', 'secondary');
+    button.textContent = 'Back to challenges';
+    button.addEventListener('click', removeModal);
```

Comment:
I normally name event handling callbacks using some well-defined scheme, to avoid confusing them with other functions. The way this code looks now, it's not clear to me when I'm reading the `removeModal()` function that it will be used as an event handler. This is risky, because it means that if I use it elsewhere, I may not realize it's also being used as en event handler, and make changes to without that understanding.

For example, imagine you also use the `removeModal()` function when clicking a new "cancel" button in step 2. You want it such that when you cancel in step 2, the code should also do some secondary action, like animate something. So you add an argument to for whether to animate, that defaults to false: `removeModal(shouldAnimate = false)`, thinking that any other call site that uses `removeModal()` without the argument should still work, because `shouldAnimate` be false, as if it wasn't there.

The problem is that event listeners always pass an event object as their first argument, so what you've done now is you've started accepting an event object and calling it `shouldAnimate`. The event object is truthy, so now the animation will try to play even when you're clicking this button here.

All because it's not obvious to the one making the changes to `removeModal()` that it needs to accommodate using it as an event handler.

In this case, I would do one of two things:

* Rename `removeModal` to `handleRemoveModal` or something like that to indicate that it is an event handler
* Keep `removeModal` defined as is, but here on line 151 do `button.addEventListener('click', () => removeModal())`, i.e. make the event handler function an anonymous function that doesn't do anything except call `removeModal()` _without any arguments_.

Reply:
EventListener always passing an object as their first argument is very important to know about! I will also go over the documentation on eventListeners too :) 

In this case I would probably do the anonymous function and run the removeModal() without arguments. I think it's more "future proof" since as you said, we might want some animation or behavior in specific situations. 

Great example of explaining why and how potential problems could arise and cause a lot of headaches!
Thanks for the feedback!